File: 58-Length-of-Last-Word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int idx = s.size() - 1;
        int length = 0;

        while(s[idx] == ' ') idx--;

        while(idx >= 0 && s[idx] != ' ') {
            length++;
            idx--;
        }

        return length;
    }
};

File: 58-Length-of-Last-Word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int n = s.length();
        int i = n-1;
        int len = 0;

        // remove leading spaces
        while (i >=0 && s.charAt(i) == ' ') {
            i--;
        }

        while (i >=0 && s.charAt(i) != ' ') {
            i--;
            len++;
        }

        return len;
        
    }
}